src/lib.rs

```diff
@@ -467,6 +492,25 @@ impl Message {
             _ => Err(Error::InvalidMessage)
         }
     }
+
+    /// Constructs a `Message` by hashing `data` with hash algorithm `H`.
+    /// ```rust
+    /// extern crate bitcoin_hashes;
+    /// use secp256k1::Message;
+    /// use bitcoin_hashes::sha256;
+    /// use bitcoin_hashes::Hash;
+    ///
+    /// let m1 = Message::from_hashed_data::<sha256::Hash>("Hello world!".as_bytes());
+    /// // is equivalent to
+    /// let m2 = Message::from(sha256::Hash::hash("Hello world!".as_bytes()));
+    ///
+    /// assert_eq!(m1, m2);
+    /// ```
+    #[cfg(feature = "bitcoin_hashes")]
+    pub fn from_hashed_data<H: ThirtyTwoByteHash + bitcoin_hashes::Hash>(data: &[u8]) -> Self {
+        let hash: H = bitcoin_hashes::Hash::hash(data);
+        Message::from(hash)
```

Comment:
nit: What a missed chance for a cool one-liner!
```rust
<H as bitcoin_hashes::Hash>::hash(data).into()
```
Don't we all love Rust! :D

Reply:
Oh yeah, that's neat, let me sneak it in :P

Reply:
Hmm, didn't notice you have the trait imported
```rust
#[cfg(feature = "bitcoin_hashes")]
use bitcoin_hashes::Hash;
```
So it could be shorted. But nvm that.

Reply:
I actually avoid that on purpose. It collides with `std::hash::Hash` too often in my experience.